File: BasicSyntacsis/advanced exercise/08.js
function pyramid(baseNum, step) {
    let stone = 0;
    let marble = 0;
    let lapisLazuli = 0;
    let gold = 0;

    let heightStep = 1;

    while (baseNum > 0) {
        if (heightStep % 5 === 0) {
            lapisLazuli += (baseNum * 4 - 4) * step;
        } else {
            marble += (baseNum * 4 - 4) * step;
        }
        stone += ((baseNum * baseNum) - (baseNum * 4 - 4)) * step;

        baseNum -= 2;
        heightStep += 1;
    }

    if (baseNum === 0) {
        baseNum = 2;
        if (heightStep % 5 === 0) {
            lapisLazuli -= (baseNum * baseNum) * step;
        } else {
            marble -= (baseNum * baseNum) * step;
        }
        gold += (baseNum * baseNum) * step;
    } else {
        stone -= (baseNum * baseNum) * step;
        gold += (baseNum * baseNum) * step;
    }


    console.log(`Stone required: ${Math.ceil(stone)}`)
    console.log(`Marble required: ${Math.ceil(marble)}`)
    console.log(`Lapis Lazuli required: ${Math.ceil(lapisLazuli)}`)
    console.log(`Gold required: ${Math.ceil(gold)}`)
    console.log(`Final pyramid height: ${Math.floor((heightStep - 1) * step)}`)
}


pyramid(23, 0.5)

File: BasicSyntacsis/advanced exercise/08.py
import math

def pyramid (base, height):
    total_stone = 0
    total_marble = 0
    total_lapis = 0
    
    step = 0

    while base > 2:
        marble = (base * 4 - 4) * height
        stone = (base * base * height) - marble

        total_stone += stone
        step += 1

        if step % 5 == 0:
            total_lapis += marble
        else:
            total_marble += marble

    step += 1

    total_gold = 1*height if step % 2 == 1 else 4*height

    print(f'Stone required: {math.ceil(total_stone)}')
    print(f'Marble required: {math.ceil(total_marble)}')
    print(f'Lapis Lazuli required: {math.ceil(total_lapis)}')
    print(f'Gold required: {math.ceil(total_gold)}')
    print(f'Final pyramid height: {math.floor(step * height)}')


pyramid (11, 1)